### ISSUE TITLE
unknown shorthand flag: 't' in -t when running devbox build

### ISSUE THREAD

**Issue by ParetoOptimalDev** [jetpack-io/devbox#28]
Docker version 20.10.12, build 20.10.12-0ubuntu4
Ubuntu 22.04.1 LTS

Since I'm getting this error I guess this doesn't use the nix docker buildTools stuff? That means it doesn't do the [docker layer optimizations nix implements](https://grahamc.com/blog/nix-and-layered-docker-images) right?

**Comment by loreto** [comment]
You are correct that we currently don't us the nix docker tools to build the image and just rely on `docker` directly. We have plans to allow for different "engines" to build the image, one of which is nix's `dockerTools` as you suggest. See https://github.com/jetpack-io/devbox/issues/18

In terms of the shorthand flag error: if you run `docker help build` does your version of docker not show `-t` as a possible flag? Does it support `--tag`? (if yes, I can easily change the code to not use the shorthand version).

**Comment by AlexDarby** [comment]
Hi, I'm also getting this message trying to run a 'devbox build'. My output of 'docker help build' shows the shortcode '-t' is available for tags.

Docker version 20.10.14, build a224086349
Ubuntu 22.04.1 LTS

devbox plan :

{
  "packages": [
    "python27"
  ]
}

**Comment by loreto** [comment]
@AlexDarby would you be able to run `docker buildx build --help` and tell me if `-t` shows up as an option there too? Sometimes we use the `buildx` engine, and I'm wondering it that's what's causing this.

**Comment by AlexDarby** [comment]
Hi @loreto , no there is no '-t' option using the `docker buildx build --help` command.

**Comment by loreto** [comment]
Got it. Ok I'll have this fixed by the next release. We should be able to make it available sometime next week.

**Comment by ParetoOptimalDev** [comment]
> Hi @loreto , no there is no '-t' option using the `docker buildx build --help` command.

All of the information is the same for me as @AlexDarby FYI.

**Comment by loreto** [comment]
Fixed is merged into `main`, it'll go out in a release by the end of the week.

**Comment by ParetoOptimalDev** [comment]
@loreto Thanks for the quick work.

 I found this issue while experimenting about blog post ideas for devbox, so this helps unblock exploring those and potentially getting one written.

**Comment by AlexDarby** [comment]
Great, thanks @loreto ! 👍

**Comment by loreto** [comment]
Closing since PR is merged. It'll be released tomorrow.
